r/src/dataset.cpp

```diff
@@ -92,6 +92,33 @@ std::shared_ptr<arrow::Schema> dataset___SFactory__Inspect(
   return VALUE_OR_STOP(factory->Inspect());
 }
 
+// DatasetFactory
+
+// [[arrow::export]]
+std::shared_ptr<ds::DatasetFactory> dataset___DFactory__Make(
+    const std::vector<std::shared_ptr<ds::SourceFactory>>& sources) {
+  return VALUE_OR_STOP(ds::DatasetFactory::Make(sources));
+}
+
+// [[arrow::export]]
+std::shared_ptr<arrow::Schema> dataset___DFactory__Inspect(
```

Comment:
You don't call Inspect so maybe just remove it?

Reply:
`Inspect` is useful for interactive schema reconciliation

Reply:
How so? Doesn't DatasetFactory->Inspect return a schema already reconciled?

Anyway, if it's useful (in R, to be clear), we should write a test showing that it works and how it might be used.

Reply:
Test written in new version, please review :)